types/seamless-immutable/seamless-immutable-tests.ts

```diff
@@ -204,8 +222,12 @@ interface ExtendedUser extends User {
     const updatedUser12: Immutable.Immutable<ExtendedUser> = immutableUserEx.setIn([ data.propertyId, 'line1' ], 'Small house');
 
     // asMutable
-    const mutableUser21: User = immutableUser.asMutable();
-    const mutableUser22: User = immutableUser.asMutable({ deep: true });
+    const mutableUser21: NonDeepMutableExtendedUser = immutableUserEx.asMutable();
```

Comment:
If possible, would you be able to update these tests so that they use the `$ExpectError` comment (see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#lint)?

We haven't using this comment in the definitions for seamless-immutable, but we probably should. Something like this:

```
// Can't mutate beyond 1 level deep
const mutableUser21 = immutableUserEx.asMutable();
mutableUser21.firstName = 'Krusty';
// $ExpectError 
mutableUser21.address.line1 = 'a line';

// Can mutate deeply beyond the first leve
const mutableUser22 = immutableUserEx.asMutable({ deep: true });
mutableUser22.firstName = 'Krusty';
mutableUser22.address.line1 = 'a line';
```

At some stage I'd like to revisit these definitions to provide better error expectations, but it would be a great start if you could add them for these couple of test cases.

Reply:
I have done my best. $ExpectsError is interesting, I didn't realize it existed.

I am pretty sure I will have to submit another PR shortly as it looks like there is an array version of set(In), getIn, update(In) (https://github.com/rtfeldman/seamless-immutable#additional-methods) and maybe others that aren't in the definitions. Currently I'm not using them but it would be nice to have them up to date. Happy to have a poke at the tests further in the same PR.